plugin/src/main/java/net/dzikoysk/funnyguilds/listener/region/PlayerMove.java

```diff
@@ -21,11 +21,13 @@ public class PlayerMove extends AbstractFunnyListener {
 
     @EventHandler
     public void onTeleport(PlayerTeleportEvent event) {
-        this.userManager.findByUuid(event.getPlayer().getUniqueId())
-                .map(User::getCache)
-                .peek(userCache -> userCache.setEnter(false));
+        if (event.getTo() != null) {
+            this.userManager.findByUuid(event.getPlayer().getUniqueId())
+                    .map(User::getCache)
+                    .peek(userCache -> userCache.setEnter(!this.regionManager.findRegionAtLocation(event.getTo()).equals(this.regionManager.findRegionAtLocation(event.getFrom()))));
 
-        this.onMove(event);
+            this.onMove(event);
+        }
```

Comment:
Lepiej odwróć warunek i zrób wcześniejszy return. Jak wołasz to dwa razy to można też wyciągnąć do zmiennej.

Reply:
+ Coś @Rollczi mówił że na nowszych wersjach to nie jest już nullem niby?

Reply:
Może i na nowych wersjach byłem nie jest null, ale warto pamiętać, że FG pamięta 1.8 i jeżeli istnieje jakikolwiek edge case to wole nie ryzykować brakiem tego ifa

Reply:
Nawet na 1.16.5 (ktore w przyszłości chcemy targetowac) jest oznaczony jako @Nullable wiec ten chceck jest potrzebny i tyle

Reply:
Dodatkowo w przypadku kiedy getTo jest null ja bym fallbackował do starego kodu gdzie setEnter bylo zawsze ustawiane na false